### PR TITLE
Add some performance configs and 3.1 fixes

### PR DESCRIPTION
**Pull request by alknopfler**
- Add performance configs
- Check some 3.1 updates to fix some version numbers

### CODE REVIEW

File: asciidoc/product/atip-automated-provision.adoc [hunk 1]
```diff
@@ -48,6 +48,16 @@ The following sections describe the different directed network provisioning work
 
 * xref:airgap-deployment[]
 
+
+[IMPORTANT NOTE]
+====
+The following sections show how to prepare the different scenarios for the directed network provisioning workflow using ATIP.
+If you want to see some examples of the configurations to deploy the different scenarios, see the following repository: https://github.com/suse-edge/atip/tree/release-3.1/telco-examples/edge-clusters[SUSE ATIP examples].
+You could find the different configurations for the `EIB` and the `Metal^3^` charts as well as some examples for the different scenarios (airgap, connected, private registry, dhcp, dhcp-less, etc).
```
Comment (atanasdinov) [inline asciidoc/product/atip-automated-provision.adoc:56]: For examples of the different configurations options for deployment (incl. air-gapped environments, DHCP and DHCP-less networks, private container registries, etc.), see the https://github.com/suse-edge/atip/tree/release-3.1/telco-examples/edge-clusters[SUSE ATIP repository].

File: asciidoc/product/atip-automated-provision.adoc [hunk 2]
```diff
@@ -270,8 +332,12 @@ When running Edge Image Builder, a directory is mounted from the host, so it is
 * `downstream-cluster-airgap-config.yaml` is the image definition file, see <<quickstart-eib>> for more details.
 * The base image when downloaded is `xz` compressed, which must be uncompressed with `unxz` and copied/moved under the `base-images` folder.
 * The `network` folder is optional, see <<add-network-eib>> for more details.
-* The `custom/scripts` directory contains scripts to be run on first-boot; currently a `01-fix-growfs.sh` script is required to resize the OS root partition on deployment. For air-gap scenarios, a script `02-airgap.sh` is required to copy the images to the right place during the image creation process.
-* The `custom/files` directory contains the `rke2` and the `cni` images to be copied to the image during the image creation process.
+* The `custom/scripts` directory contains scripts to be run on first-boot:
+    1. `01-fix-growfs.sh` script is required to resize the OS root partition on deployment.
+    2. For air-gap scenarios, a script `02-airgap.sh` is required to copy the images to the right place during the image creation process.
```
Comment (atanasdinov) [inline asciidoc/product/atip-automated-provision.adoc:337]: To keep up the consistency:
```suggestion
    2. `02-airgap.sh` script is required to copy the images to the right place during the image creation process for air-gapped environments.
```

File: asciidoc/product/atip-features.adoc [hunk 3]
```diff
@@ -60,6 +61,38 @@ In our case, if you have installed a real-time image like `SLE Micro RT`, kernel
 For more information about the real-time kernel, visit https://www.suse.com/products/realtime/[SUSE Real Time].
 ====
 
+[#kernel-args]
+=== Kernel arguments for low latency and high performance
+
+The kernel arguments are important to be configured to enable the real-time kernel to work properly giving the best performance and low latency to run telco workloads.  There are some important concepts to keep in mind when configuring the kernel arguments for this use case:
+
+* Remove `kthread_cpus` when using SUSE kernel real time. This parameter controls on which CPUs kernel threads are created. It also controls which CPUs are allowed for PID 1 and for loading kernel modules (the kmod user-space helper). This parameter is not
```
Comment (atanasdinov) [inline asciidoc/product/atip-features.adoc:69]:
```suggestion
* Remove `kthread_cpus` when using SUSE real-time kernel. This parameter controls on which CPUs kernel threads are created. It also controls which CPUs are allowed for PID 1 and for loading kernel modules (the kmod user-space helper). This parameter is not
```

File: asciidoc/product/atip-features.adoc [hunk 4]
```diff
@@ -60,6 +61,38 @@ In our case, if you have installed a real-time image like `SLE Micro RT`, kernel
...
+In contrast, `idle=poll` runs in a tight loop, busy-waiting for a task to be rescheduled. This minimizes the latency of exiting the idle state, but at the cost of keeping the CPU running at full speed in the idle thread.
+
+* Disable C1E in BIOS. This option is important to disable the C1E state in the BIOS to avoid the CPU from entering the C1E state when idle. The C1E state is a low-power state that can introduce latency when the CPU is idle.
+
+* Add `nowatchdog` to disable the soft-lockup watchdog which is implemented as a timer running in the timer hard-interrupt context. When it expires (i.e. a soft lockup is detected), it will print a warning (in the hard interrupt context), running any latency targets. Even if it never expires, it goes onto the timer list, slightly increasing the overhead of every timer interrupt.
+This option also disables the NMI watchdog, so NMIs cannot interfere.
+
+* Add `nmi_watchdog=0`. This option disables only the NMI watchdog.
+
+With the values shown above an example of the kernel arguments to be used is shown below:
```
Comment (atanasdinov) [inline asciidoc/product/atip-features.adoc:88]:
```suggestion
This is an example of the kernel argument list including the aforementioned adjustments:
```

File: asciidoc/product/atip-requirements.adoc [hunk 5]
```diff
@@ -106,3 +108,22 @@ rebootmgrctl strategy off
...
+----
+
+* `fstrim` is a service that allows to trim the filesystems automatically every week. For Telco workloads, it is important to disable the automatic trim to avoid any impact on the services running on the nodes.
+
+To disable the automatic trim, you can run:
+
+[,shell]
+----
+systemctl --now disable fstrim.timer
+----
```
Comment (atanasdinov) [inline asciidoc/product/atip-requirements.adoc:129]: Aren't these part of the EIB config?

File: asciidoc/product/atip-automated-provision.adoc [hunk 6]
```diff
@@ -337,6 +407,41 @@ cp install.sh /opt/
 cp rke2-images*.tar.zst rke2.linux-amd64.tar.gz sha256sum-amd64.txt /opt/rke2-artifacts/
 ----
 
+[#add-custom-script-performance2]
+===== Performance script
+
+The following optional script (`custom/scripts/03-performance.sh`) can be used to configure the system for performance tuning:
```
Comment (atanasdinov) [inline asciidoc/product/atip-automated-provision.adoc:413]: I don't think the content of these scripts is relevant for the reader. It's about getting the files to the filesystem and it is rather an implementation detail but I'd let @hardys chime in on this.

Perhaps in the future we can use the new [OS files](https://github.com/suse-edge/edge-image-builder/blob/main/docs/building-images.md#operating-system-files) functionality of EIB and simplify this.

File: asciidoc/product/atip-automated-provision.adoc (hunk 1 above)
Comment (atanasdinov) [inline asciidoc/product/atip-automated-provision.adoc:56]:
```suggestion
For examples of the different configurations options for deployment (incl. air-gapped environments, DHCP and DHCP-less networks, private container registries, etc.), see the https://github.com/suse-edge/atip/tree/release-3.1/telco-examples/edge-clusters[SUSE ATIP repository].
```

File: asciidoc/product/atip-automated-provision.adoc (hunk 2 above)
Comment (alknopfler) [reply asciidoc/product/atip-automated-provision.adoc]: changed

File: asciidoc/product/atip-features.adoc (hunk 3 above)
Comment (alknopfler) [reply asciidoc/product/atip-features.adoc]: changed

File: asciidoc/product/atip-features.adoc (hunk 4 above)
Comment (alknopfler) [reply asciidoc/product/atip-features.adoc]: changed

File: asciidoc/product/atip-requirements.adoc (hunk 5 above)
Comment (alknopfler) [reply asciidoc/product/atip-requirements.adoc:129]: This is the non-automated workflow, so in this part of the documentation we're explaining the manual way to configure most of the telco capabilities we're automating in the following section

File: asciidoc/product/atip-automated-provision.adoc (hunk 6 above)
Comment (alknopfler) [reply asciidoc/product/atip-automated-provision.adoc:412]: I think just to make it consistent we need to show it until we can automate it with the os files as you said, but we must show the script content to explain it by now.

File: asciidoc/product/atip-automated-provision.adoc (hunk 1 above)
Comment (alknopfler) [reply asciidoc/product/atip-automated-provision.adoc]: changed

**Review by atanasdinov — APPROVED** [review]

**Pull request merged**